src/xtopcom/xcontract_runtime/src/xtop_action_generator.cpp

```diff
@@ -14,12 +14,26 @@ NS_BEG2(top, contract_runtime)
 
 std::unique_ptr<data::xbasic_top_action_t const> xtop_action_generator::generate(xobject_ptr_t<data::xcons_transaction_t> const & tx) {
     common::xaccount_address_t const target_address{ tx->get_transaction()->get_target_addr() };
+    
+    // eth deploy code
+    if (target_address.empty()) {
+        common::xaccount_address_t const source_address{ tx->get_transaction()->get_source_addr() };
+        if (source_address.type() != base::enum_vaccount_addr_type_secp256k1_evm_user_account) {
+            assert(false);
+            return nullptr;
```

Comment:
looks like we need add an `ec` parameter?

Reply:
right